### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by citruscaviar**

### CODE REVIEW

File: src/main/java/com/onlinebookstore/store/repository/BookRepository.java [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
+import org.springframework.stereotype.Repository;
 
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> findAll();
-
-    Optional<Book> findById(Long id);
+@Repository
+public interface BookRepository extends JpaRepository<Book,Long> {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/onlinebookstore/store/repository/BookRepository.java:8]:
```suggestion
public interface BookRepository extends JpaRepository<Book, Long> {
```

File: src/main/java/com/onlinebookstore/store/service/BookServiceImpl.java [hunk 2]
```diff
@@ -38,12 +36,21 @@ public BookDto findById(Long id) {
...
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Couldn't find a book with id: " + id));
+        book.setTitle(requestDto.getTitle());
+        book.setAuthor(requestDto.getAuthor());
+        book.setIsbn(requestDto.getIsbn());
+        book.setPrice(requestDto.getPrice());
+        book.setDescription(requestDto.getDescription());
+        book.setCoverImage(requestDto.getCoverImage());
```
Comment (liliia-ponomarenko) [inline src/main/java/com/onlinebookstore/store/service/BookServiceImpl.java:48]: you should use Mapper for this logic.

File: src/main/resources/db/changelog/changes/02-fill-books-table.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                  value: "9781122334455"
+              - column:
+                  name: "price"
+                  value: "29.99"
+              - column:
+                  name: "description"
+                  value: "This is a sample book description."
+              - column:
+                  name: "cover_image"
+                  value: "https://example.com/cover3.jpg"
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/02-fill-books-table.yaml:68]: each file must end with an empty line (check other files)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution ;)

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                  type: VARCHAR(1024)
+              - column:
+                  name: cover_image
+                  type: VARCHAR(255)
+              - column:
+                  name: is_deleted
+                  type: BOOLEAN
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: "false"
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml:48]: add empty line at the end of the file

File: src/main/resources/db/changelog/changes/02-fill-books-table.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
+databaseChangeLog:
+  - changeSet:
+      id: fill-books-table
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-fill-books-table.yaml:3]: you don't need to save books this way, remove changeset

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
+  - include:
+      file: db/changelog/changes/02-fill-books-table.yaml
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/db.changelog-master.yaml:5]: add empty line at the end of the file

File: src/test/resources/application.properties [hunk 7]
```diff
@@ -3,3 +3,7 @@ spring.datasource.driverClassName=org.h2.Driver
 spring.datasource.username=sa
 spring.datasource.password=password
 spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+spring.jpa.hibernate.ddl-auto=validate
+spring.jpa.show-sql=true
+spring.jpa.open-in-view=false
+spring.jackson.deserialization.fail-on-unknown-properties=true
```
Comment (Elena-Bruyako) [inline src/test/resources/application.properties:9]: add empty line at the end of the file

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/onlinebookstore/store/mapper/BookMapper.java [hunk 8]
```diff
@@ -6,11 +6,15 @@
...
+import org.mapstruct.MappingTarget;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
     BookDto toDto(Book book);
 
     @Mapping(target = "id",ignore = true)
     Book toModel(CreateBookRequestDto requestDto);
+
+    @Mapping(target = "id", ignore = true) // Ignore id when updating
```
Comment (liliia-ponomarenko) [inline src/main/java/com/onlinebookstore/store/mapper/BookMapper.java:18]:
```suggestion
    @Mapping(target = "id", ignore = true) 
```
redundant comment

File: src/main/resources/application.properties [hunk 9]
```diff
@@ -1,7 +1,11 @@
...
-spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+server.servlet.context-path=/api
 
-spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQLDialect
+spring.jpa.open-in-view=false
+
```
Comment (liliia-ponomarenko) [inline src/main/resources/application.properties:11]:
```suggestion
```
only one empty line at the end of the file ;)

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! But please fix my comment before merging ;)

**Pull request merged**